types/src/system/auction/mod.rs

```diff
@@ -412,9 +416,15 @@ pub trait Auction:
         };
 
         // Increment era
-        era_id += 1;
+        era_id = match era_id.checked_add(1) {
+            Some(era_id) => era_id,
+            None => return Err(Error::ArithmeticOverflow),
+        };
```

Comment:
```suggestion
        era_id = era_id.checked_add(1).ok_or(Error::ArithmeticOverflow)?;
```
A similar change should be done to the 7 other instances in this file.

Reply:
Applied suggestion in the 7 other places as well